user-guide/Advanced_Functionality/DataMiner_Agents/Upgrading_a_DMA/Preparing_to_upgrade_a_DataMiner_Agent.md

```diff
@@ -0,0 +1,67 @@
+
+1. If the upload was successful, you should receive the following upload report:
+
+   ```txt
+   Upload report
+   Upload has been completed successfully.
+   ```
+
+> [!TIP]
+> Upload the package at least a week before executing the upgrade.
+
+## Prerequisites
+
+Prerequisites are small self-contained apps that run during the upload phase of a DataMiner upgrade. Their purpose is to detect whether all necessary conditions for upgrading DataMiner to the selected version and all requirements for the DataMiner Agent to run are met. If the prerequisites detect that this is not the case, the upgrade will be aborted.
+
+As of now, the following prerequisites are available:
+
+- Verify Cluster Ports: Verifies whether the ports used by DataMiner can be reached in between DataMiner Agents. If this check fails, you will need to install the [VerifyClusterPorts.dmupgrade](xref:VerifyClusterPortsdmupgrade) package. From DataMiner 10.2.0 [CU2]/10.2.5 onwards, this prerequisite is available by default and runs automatically when you upgrade.
+
+- [Verify NATS is Running](xref:VerifyNatsIsRunning): Verifies whether the crucial NATS service used by DataMiner is running on all required DataMiner Agents. From DataMiner 10.2.0 [CU14]/10.2.7 onwards, this prerequisite is available by default and runs automatically when you upgrade.
+
+- [Verify Cloud DxM Version](xref:BPA_Verify_Cloud_DxM_Version): Verifies whether the minimum required version is installed for all DxMs in the system. From DataMiner 10.2.0 [CU6]/10.2.8 onwards, this prerequisite is available by default and runs automatically when you upgrade.
+
+- Service Automatic Properties
+
+- [Validate Connectors](xref:BPA_Validate_Connectors): Scans the DataMiner System for any connectors that are known to be incompatible with the DataMiner version to which the DataMiner Agent is being upgraded. From DataMiner 10.3.4/10.4.0 onwards, this prerequisite is available by default and runs automatically when you upgrade.
+
+## Having a backup at the ready
+
+We recommend making a backup of your DataMiner System before executing an upgrade. In case unexpected issues occur after an update, downgrading may be required.
+
+Making a backup can be done with a VM snapshot or DataMiner.
+
+### VM snapshot
+
+[Take a virtual machine (VM) snapshot](xref:MOP_Downgrading_a_DMS#for-a-downgrade-with-vm-snapshot-restore) of the upgraded machines. This will allow a speedy rollback of the DataMiner system.
+
+> [!TIP]
+> For more information on how to downgrade your DMS using a VM snapshot restore, see [Downgrade with VM snapshot restore](xref:MOP_Downgrading_a_DMS#downgrade-with-vm-snapshot-restore-preferred).
+
+> [!NOTE]
+> The timing for a VM snapshot restore depends on your specific setup.
+
+### DataMiner backup
+
+To make a backup of a DataMiner Agent using DataMiner Taskbar Utility or DataMiner Cube, see [Backing up a DataMiner Agent](xref:Backing_up_a_DataMiner_Agent).
```

Comment:
For the sake of continuity, and if this is useful here, we could perhaps add a tip with more information on how to downgrade a DMS using a DataMiner backup as well? I am however unsure of whether to add a link to [Restoring a DataMiner Agent](https://docs.dataminer.services/user-guide/Advanced_Functionality/DataMiner_Agents/Restoring_a_DataMiner_Agent.html) or to [Downgrading a DMS](https://docs.dataminer.services/user-guide/Reference/MOPs/MOP_Downgrading_a_DMS.html).

Reply:
I've deleted all mentions to downgrading a DMS, as this was a misinterpretation on my part that could indeed be confusing and a no-go in the case of a DataMiner backup. See [commit](https://github.com/SkylineCommunications/dataminer-docs/pull/1188/commits/a339ad7ec18dde526354d2361490d41cfa2b0cc1).